src/test/scala/org/broadinstitute/dsde/firecloud/webservice/PermissionReportApiSpec.scala

```diff
@@ -0,0 +1,260 @@
+package org.broadinstitute.dsde.firecloud.webservice
+
+import org.broadinstitute.dsde.firecloud.FireCloudExceptionWithErrorReport
+import org.broadinstitute.dsde.firecloud.dataaccess.{MockAgoraDAO, MockRawlsDAO}
+import org.broadinstitute.dsde.firecloud.model.ErrorReportExtensions.FCErrorReport
+import org.broadinstitute.dsde.firecloud.model.MethodRepository._
+import org.broadinstitute.dsde.firecloud.model.{MethodConfigurationName, PermissionReport, PermissionReportRequest, UserInfo, WithAccessToken}
+import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol._
+import org.broadinstitute.dsde.firecloud.service.{BaseServiceSpec, PermissionReportService, WorkspaceService}
+import org.broadinstitute.dsde.rawls.model.{MethodConfigurationShort, MethodRepoMethod, _}
+import org.scalatest.BeforeAndAfterEach
+import spray.http.{HttpMethods, HttpResponse}
+import spray.http.StatusCodes._
+import spray.httpx.SprayJsonSupport._
+import spray.json.DefaultJsonProtocol._
+
+import scala.concurrent.Future
+
+class PermissionReportApiSpec extends BaseServiceSpec with WorkspaceApiService with BeforeAndAfterEach {
+
+  def actorRefFactory = system
+
+  val testApp = app.copy(agoraDAO=new PermissionReportMockAgoraDAO(), rawlsDAO=new PermissionReportMockRawlsDAO())
+
+  val workspaceServiceConstructor: (WithAccessToken) => WorkspaceService = WorkspaceService.constructor(testApp)
+  val permissionReportServiceConstructor: (UserInfo) => PermissionReportService = PermissionReportService.constructor(testApp)
+
+  def permissionReportPath(ns:String,name:String) = s"/api/workspaces/$ns/$name/permissionReport"
+
+  "Permission Report API" - {
+
+    "should only accept POST" in {
+      List(HttpMethods.GET, HttpMethods.PUT, HttpMethods.PATCH, HttpMethods.DELETE) map {
+        method =>
+          new RequestBuilder(method)(permissionReportPath("foo","bar")) ~> dummyUserIdHeaders("1234") ~> sealRoute(workspaceRoutes) ~> check {
+            status should equal(MethodNotAllowed)
+          }
+      }
+    }
+
+    "should reject malformed input" in {
+      // endpoint expects an object; send it an array
+      val badPayload = Seq("one","two","three")
```

Comment:
While this test works, the API will still respond to bad input in a different form. If the item is a json object, then the API just ignores the bad elements (since they're options) and treats it as an empty json object. Not sure if this is expected/desired behavior. Personally, I like it as it is more forgiving of user errors. For example, this returns all users and all referenced methods in the report:
```
curl -X POST --header 'Content-Type: application/json' \
    --header 'Accept: application/json' --header 'Authorization: Bearer blah' \
    -d '{"one":"two", "three": "four"}' \
    'https://local.broadinstitute.org:10443/api/workspaces/broad-dsde-dev/grAddTest/permissionReport'
```